### PR TITLE
luci-app-olsrd2: New Package for OLSR2 configuration and status visualisation

### PR DESCRIPTION
**Pull request by stargieg**
Compile tested: mips_24kc, arm_cortex-a9_vfpv3-d16, i386_pentium4, x86_64, i386_pentium-mmx, mipsel_24kc

Signed-off-by: Patrick Grimm <patrick@lunatiki.de>

### CODE REVIEW

**Comment by mwarning** [comment]
@stargieg thank you for you effort. Is this code from a different repository?
@HRogge @PolynomialDivision please have a look.

**Comment by stargieg** [comment]
Yes. It come from https://github.com/stargieg/luci2-ffwizard/commits/master/luci-app-olsr2
I have it renamed.

**Comment by BKPepe** [comment]
This package should be most likely moved to the [LuCI repository](https://github.com/openwrt/luci), where it will be properly reviewed and taken care by LuCI developers and its community, not here, where you can find only routing packages.

File: luci-app-olsrd2/Makefile [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# call BuildPackage - OpenWrt buildroot signature
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=luci-app-olsrd2
+PKG_VERSION:=0.2.6
+PKG_RELEASE:=14
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
```
Comment (BKPepe) [inline luci-app-olsrd2/Makefile:7]: Drop, this is alrealdy default

File: luci-app-olsrd2/Makefile [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# call BuildPackage - OpenWrt buildroot signature
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=luci-app-olsrd2
+PKG_VERSION:=0.2.6
+PKG_RELEASE:=14
```
Comment (BKPepe) [inline luci-app-olsrd2/Makefile:6]: This is new package here, it should start with 1.

File: luci-app-olsrd2/Makefile [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/luci-app-olsrd2
+  SECTION:=luci
+  CATEGORY:=LuCI
+  SUBMENU:=3. Applications
+  TITLE:=OLSR2 configuration and status module
+  MAINTAINER:=Patrick Grimm <patrick@lunatiki.de>
+  EXTRA_DEPENDS:=oonf-olsrd2, luci-mod-admin-full
```
Comment (BKPepe) [inline luci-app-olsrd2/Makefile:17]: Extra depends? Why? This package depends on ``oonf-olsrd2``. It's not extra dependency.
I wonder, why do you need to have ``luci-mod-admin-full``?

File: luci-app-olsrd2/Makefile [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+define Build/Prepare
+endef
+
+define Build/Configure
+endef
+
+define Build/Compile
+endef
+
+define Package/$(PKG_NAME)/postinst
```
Comment (BKPepe) [inline luci-app-olsrd2/Makefile:30]: You are reusing ``PKG_NAME`` variable, which was removed by treewide commit in packages repository several years ago. It makes things harder to review.

So you should use here:
define Package/luci-app-olsrd2/postinst

File: luci-app-olsrd2/Makefile [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+#!/bin/sh
+if [ -z $${IPKG_INSTROOT} ] ; then
+	rm -f /tmp/luci-indexcache
+	rm -rf /tmp/luci-modulecache/
+	killall -HUP rpcd 2>/dev/null
```
Comment (BKPepe) [inline luci-app-olsrd2/Makefile:35]: I don't think that this is a good solution.

File: luci-app-olsrd2/root/etc/uci-defaults/41_luci-olsrd2 [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+uci -q batch <<-EOF >/dev/null
+	delete ucitrack.@olsrd2[-1]
+	add ucitrack olsrd2
+	set ucitrack.@olsrd2[-1].init=olsrd2
+	commit ucitrack
+EOF
+
+rm -f /tmp/luci-indexcache
+rm -f /tmp/luci-modulecache/*
```
Comment (BKPepe) [inline luci-app-olsrd2/root/etc/uci-defaults/41_luci-olsrd2:11]: Why? This was already removed by treewide commit in LuCI and you are adding it back
- https://github.com/openwrt/luci/commit/1f8a3c04178e21f45f3908fd476ec3b3dad73c69

File: luci-app-olsrd2/root/etc/config/luci_olsrd2 [hunk 7]
```diff
@@ -0,0 +1,3 @@
+config 'olsrd2' 'general'
+	option 'resolve' '1'
+	option 'domain' 'olsr'
```
Comment (BKPepe) [inline luci-app-olsrd2/root/etc/config/luci_olsrd2:3]: This looks to me that it should not be located in LuCI package. Yes, you can find some a few outdated packages with LuCI, which comes with config file, but thats not common these days. Its not marked even as conffile, though.

**Pull request merged**